src/codegen/compile.ml

```diff
@@ -7292,7 +7062,7 @@ module MakeSerialization (Strm : Stream) = struct
       (* returns true if we are looking at primitive type with this id *)
       let check_prim_typ t =
         get_idltyp ^^
-        compile_eq_const (Int32.neg (Option.get (to_idl_prim t)))
+        compile_eq_const (Int32.neg (Option.get (to_idl_prim Candid t)))
```

Comment:
`Candid` looks incorrect. This should probbly be parameter depending on the kind of deserialization we are doing.
I don't think we can easily share generated candid/persistence code.

Reply:
Actually, this is fine after all.